solaris/vector/mask.py

```diff
@@ -945,6 +945,17 @@ def instance_mask(df, out_file=None, reference_im=None, geom_col='geometry',
     for idx, feat in enumerate(feature_list):
         output_arr[:, :, idx] = features.rasterize([feat], out_shape=shape,
                                                    transform=affine_obj)
+
+    if reference_im:
+        reference_im = _check_rasterio_im_load(reference_im)
+        try:
+            bad_data_mask = (reference_im.read() == reference_im.nodata).any(axis=0) # take logical and along all dims so that all pixxels not -9999 across bands
```

Comment:
I have a minor change to how this try/except loop works. Given that there _may_ be some situations where users might not want to kill their entire pipeline if they don't have a nodata value at this line, but rather catch the exception and do something, I'd rather use this approach:

```suggestion
    try:
            bad_data_mask = (reference_im.read() == reference_im.nodata).any(axis=0) # take logical and along all dims so that all pixxels not -9999 across bands
    except AttributeError as ae:  # raise another, more verbose AttributeError
        raise AttributeError("A nodata value is not defined for the source image. Make sure the reference_im has a nodata value defined.") from ae
```

That would result in something like the following when the exception is thrown:
```
Traceback (most recent call last):
  File "mask.py", line 952, in df_to_px_mask()
    (more trace details here)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "mask.py", line 955, in df_to_px_mask()
    raise AttributeError("A nodata value is not defined for the source image. Make sure the reference_im has a nodata value defined.") from e
AttributeError: A nodata value is not defined for the source image. Make sure the reference_im has a nodata value defined.
```